src/components/policy/policy_external/src/cache_manager.cc

```diff
@@ -642,12 +645,22 @@ void CacheManager::ProcessUpdate(
         initial_policy_iter) {
   using namespace policy;
   using rpc::policy_table_interface_base::RequestTypes;
+  using rpc::policy_table_interface_base::ApplicationParams;
   const RequestTypes& new_request_types =
       *(initial_policy_iter->second.RequestType);
 
   const std::string& app_id = initial_policy_iter->first;
   bool update_request_types = true;
 
+  ApplicationParams& params =
+      pt_->policy_table.app_policies_section.apps[app_id];
+  if (kPreDataConsentId == app_id) {
+    *(params.heart_beat_timeout_ms) =
+        *(initial_policy_iter->second.heart_beat_timeout_ms);
+    LOG4CXX_INFO(logger_,
+                 "heart_beat_timeout_ms in predata = "
+                     << *(params.heart_beat_timeout_ms));
+  }
```

Comment:
I think these lines are unnecessary, it will print the heartbeat timeout every PTU. I think it is enough to just print the heartbeat every time it is updated like you do in `SetHeartBeatTimeout`